### PR TITLE
Fixes #25312 - skip add_*permissions_to_default_roles in rake

### PR DESCRIPTION
**Pull request by evgeni**

### CODE REVIEW

File: app/registries/foreman/plugin.rb [hunk 1]
```diff
@@ -294,6 +294,7 @@ def role(name, permissions, description = '')
     # Usage:
     # add_resource_permissions_to_default_roles ['MyPlugin::FirstResource', 'MyPlugin::SecondResource'], :except => [:skip_this_permission]
     def add_resource_permissions_to_default_roles(resources, opts = {})
+      return if Foreman.in_setup_db_rake? || !permission_table_exists?
```
Comment (stbenjam) [inline app/registries/foreman/plugin.rb:297]: Would it be better for this early return to go in RbacSupport, i.e. where the actual DB activity is going to happen?

Comment (stbenjam) [reply app/registries/foreman/plugin.rb:297]: Although there's other early returns in this plugin code, so maybe here is best. :man_shrugging:

Comment (evgeni) [reply app/registries/foreman/plugin.rb:297]: If you look at the history of this file, that (other) return was moved here from Rbac some time ago.

**Comment by evgeni** [comment]
[test foreman]

**Comment by evgeni** [comment]
[test foreman]

**Review by tbrisker — APPROVED** [review]
Thanks @evgeni

**Comment by tbrisker** [comment]
Do we need this in 1.20?

**Comment by evgeni** [comment]
No plugins use the codepath today. Katello would be the first with https://github.com/Katello/katello/pull/7532

So I'd say: doesn't hurt, but is also not really required.

**Pull request merged**